Library/Homebrew/download_strategy.rb

```diff
@@ -282,6 +283,14 @@ def initialize(name, resource)
   end
 
   def fetch
+    if ENV["HOMEBREW_ARTIFACT_DOMAIN"]
+      if @url.include?(BottleSpecification::DEFAULT_DOMAIN)
+        @url.sub!(BottleSpecification::DEFAULT_DOMAIN, ENV["HOMEBREW_ARTIFACT_DOMAIN"])
+      else
+        @url = @url.sub(/^((ht|f)tps?:\/\/)?/, "#{ENV["HOMEBREW_ARTIFACT_DOMAIN"]}/#{Utils::Bottles::Bintray.repository(tap)}/")
```

Comment:
#CASE 1: 
###HOMEBREW_ARTIFACT_DOMAIN is set and formula is bottled

export HOMEBREW_ARTIFACT_DOMAIN=http://localhost:8080/artifactory

When user installs formula 

```brew install node```

Downloading http://localhost:8080/artifactory/bottles/node-6.2.2.el_capitan.bottle.tar.gz

#CASE 2: 
###HOMEBREW_ARTIFACT_DOMAIN is set and formula is non-bottled

export HOMEBREW_ARTIFACT_DOMAIN=http://localhost:8080/artifactory

When user installs formula 

```brew install riak```

Downloading http://localhost:8080/artifactory/bottles/s3.amazonaws.com/downloads.basho.com/riak/2.1/2.1.4/osx/10.8/riak-2.1.4-OSX-x86_64.tar.gz

Note: For non-bottled formulas, it replaces Protocol with HOMEBREW_ARTIFACT_DOMAIN